Describe fingerprinting by counting number of mdns candidates (#55)
youennf/mdns-ice-candidates: draft-ietf-rtcweb-mdns-ice-candidates.md

has a different origin than the top-level browsing context), or a private
browsing context.

Network Interface Enumeration
----------------------------

> The number of mDNS hostname candidates might provide a fingerprinting dimension.

Reviewer: ```suggestion
Even when local IP addresses are not exposed, the number of mDNS hostname candidates can still provide a fingerprinting dimension.
```

> This is in particular the case for network interfaces whose existence are not exposed

Reviewer: ```suggestion
This is in particular the case for network interfaces that do not have Internet connectivity and will not generate server-reflexive or relay candidates.
```

through other means like STUN or TURN servers.

> No new information is provided when restricting mDNS hostname candidates to default route only.

Reviewer: Some estimate of the amount of entropy surfaced here would help readers assess the importance of this finding.

Author: Agreed, and hopefully it should be small.
I am unsure how we can evaluate the amount of entropy without doing some experiments there.

The main point though is to say that we should only care about this at the time we will consider exposing non default route candidates.

Reviewer: I think this document should probably stick to the technique and let other documents make recommendations about modes, so we probably need to have some short, if hand-wavy, discussion of impact.

> No new information is provided when restricting mDNS hostname candidates to default route only.

Reviewer: ```suggestion
Note that no new information is exposed when mDNS candidates are only provided for the default route.
```

has a different origin than the top-level browsing context), or a private
browsing context.

Network Interface Enumeration
-----------------------------

Even when local IP addresses are not exposed, the number of mDNS hostname
candidates can still provide a fingerprinting dimension. This is in particular
the case for network interfaces with limited connectivity that will not generate

> any server-reflexive or relay candidate.

Reviewer: ```suggestion
server-reflexive or relay candidates.
```

The more mDNS names an endpoint exposes through mDNS hostname candidates, the

> higher is the fingerprinting risk. One countermeasure is to limit this number to

Reviewer: ```suggestion
higher the fingerprinting risk. One countermeasure is to limit this number to
```

> a small value. Note also that exposing only default route candidates

Reviewer: I liked the previous formulation better, e.g.

> Note that no additional fingerprinting risk is introduced when restricting mDNS hostname candidates to default route only.

This is probably best as its own paragraph, but could be the final sentence in this paragraph.

Network Interface Enumeration
-----------------------------

Even when local IP addresses are not exposed, the number of mDNS hostname
candidates can still provide a fingerprinting dimension. This is in particular
the case for network interfaces with limited connectivity that will not generate
any server-reflexive or relay candidate.

> Ideally, the number of discoverable server-reflexive candidates will be equal

Reviewer: Not sure I follow; how is this possible? Wouldn't we always have equal or fewer srflx candidates?

Reviewer 2: Relative to host candidates, yes, but @youennf wants to limit the derived mDNS candidates. If we expose more mDNS candidates than srflx candidates, we're leaking that there are (private) host candidates in addition to the ones used to create the srflx ones, giving away a little bit of private information?

Reviewer: Ah, I see what you're getting at. Maybe we should leave this for later, as I can think of a couple issues that this could cause:
- no candidates at all if a STUN server isn't provided or isn't reachable
- host candidates aren't exposed until the STUN check comes back, increasing latency

Author: Right, the idea was to describe in terms of total number of "potential" server-reflexive candidates, not in terms of implementation strategy. Maybe this criteria is not great.
I'll do another try at it.
